dbms/src/Flash/Coprocessor/DAGExpressionAnalyzer.cpp

```diff
@@ -158,6 +158,36 @@ void DAGExpressionAnalyzer::fillArgumentDetail(
     arg_collators.push_back(removeNullable(arg_types.back())->isString() ? getCollatorFromExpr(arg) : nullptr);
 }
 
+void DAGExpressionAnalyzer::fillArgumentDetailForAggFuncBitwise(
+    const ExpressionActionsPtr & actions,
+    const tipb::Expr & arg,
+    Names & arg_names,
+    DataTypes & arg_types,
+    TiDB::TiDBCollators & arg_collators)
+{
+    DataTypePtr uint64_type = std::make_shared<DataTypeUInt64>();
+    const Block & sample_block = actions->getSampleBlock();
+    String name = getActions(arg, actions);
+    DataTypePtr orig_type = sample_block.getByName(name).type;
+    // Bitwise operations can only be applied to integer types.
+    // So we need to cast the argument to UInt64.
```

Comment:
why we need to add cast here?
Didn't tidb add a cast here?

Reply:
No. We compute the result in TiFlash, so we should cast in TiFlash.

Reply:
It seems tidb does type cast during operator execution phase.
https://github.com/pingcap/tidb/blob/bce6901705cde161451c0bd707a07ecfac0c7578/expression/aggregation/bit_or.go#L37-L56. This behavior is kind of similar to non-agg bitwise function. I think we might have two choice

1. also do type cast during operator execution phase instead of appending cast beforehand
2. do the same as non-agg bitwise function, i.e. appending cast in advance

https://github.com/pingcap/tiflash/blob/bcb029eaa1e0bc616b279dfc37232e823f987d76/dbms/src/Flash/Coprocessor/DAGExpressionAnalyzerHelper.cpp#L326-L357

However, I think current comment is not accurate. TiDB always pushes integer types to tiflash. The reason why we append a cast, or do type cast in execution phase, is because mysql (prior to 8.0) handle only unsigned 64-bit integer argument and result values.

ref: https://dev.mysql.com/doc/refman/8.0/en/bit-functions.html